trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/req2pea/ReqCheckAnnotator.java

```diff
@@ -171,8 +186,54 @@ private List<Statement> generateAnnotations() {
 			annotations.addAll(genChecksNonVacuity(mUnitLocation));
 		}
 		annotations.addAll(genChecksRTInconsistency(mUnitLocation));
+		if(mCheckStateRecoverability) {
+			annotations.addAll(genCheckStateRecoverability(mUnitLocation));
+		}
+		
 		return annotations;
 	}
+	
+	private List<Statement> genCheckStateRecoverability(final BoogieLocation bl) {
+		List<Statement> list = new ArrayList<>();
+		StateRecoverabilityGenerator mStRecGen = new StateRecoverabilityGenerator();
+		AuxStatementContainer auxStatementContainer = mSymbolTable.getAuxStatementContainer();
+		Map<VerificationExpression, Map<PhaseEventAutomata, Set<StateRecoverabilityAuxStatement>>> vePeaAuxStatementMap = mStRecGen.getAuxStatementPerVerificationExpression(auxStatementContainer);
+		
+		for(Map.Entry<VerificationExpression, Map<PhaseEventAutomata, Set<StateRecoverabilityAuxStatement>>> entry : vePeaAuxStatementMap.entrySet()) {
+			VerificationExpression ve = entry.getKey();
+			
+			for(Map.Entry<PhaseEventAutomata, Set<StateRecoverabilityAuxStatement>> entryPeaStRecAuxSt : entry.getValue().entrySet()) {
+					Set<StateRecoverabilityAuxStatement> stRecAuxStSet = entryPeaStRecAuxSt.getValue();
+					for(StateRecoverabilityAuxStatement stRecAuxSt : stRecAuxStSet) {
+					String checkLabel = "STATE_RECOVERABILITY_" + entry.getKey().getVariable() + "_" + stRecAuxSt.getPcVariable();
+					
+					Expression globalVariableTrue = mStRecGen.createExpression(bl, BoogieType.TYPE_BOOL, stRecAuxSt.getRelatedVariable(), Operator.COMPEQ, "true");
+					Expression inputCondition = mStRecGen.createExpression(bl, BoogiePrimitiveType.toPrimitiveType(ve.getDataType()), ve.getVariable(), ve.getOperator(), ve.getValue());
+					Expression andCondition = ExpressionFactory.newBinaryExpression(bl, Operator.LOGICAND, globalVariableTrue, inputCondition);
+					Expression expr = ExpressionFactory.constructUnaryExpression(bl, UnaryExpression.Operator.LOGICNEG, andCondition);
+					
+					final ReqCheck check = createReqCheck(Spec.STATE_RECOVERABILITY, stRecAuxSt.getPeaPhasePc().getReq(), entryPeaStRecAuxSt.getKey(), String.join("", ve.getExpr()));
+					list.add(createAssert(expr, check, checkLabel));
+				}
+			}
+		}
+		
+		// DOT printen
+		/*
+		final List<Entry<PatternType<?>, PhaseEventAutomata>> counterTracePEAList = new ArrayList<>();
+		for (final ReqPeas reqPea : mReqPeas) {
+			final PatternType<?> pattern = reqPea.getPattern();
+			
+			for (final Entry<CounterTrace, PhaseEventAutomata> pea : reqPea.getCounterTrace2Pea()) {
+				counterTracePEAList.add(new Pair<>(pattern, pea.getValue()));
+			}
+		}
+		final List<Entry<PatternType<?>, PhaseEventAutomata>[]> subsets = CrossProducts.subArrays(counterTracePEAList.toArray(new Entry[counterTracePEAList.size()]), counterTracePEAList.size(), new Entry[counterTracePEAList.size()]);
+		//Prints parallel automaton
+		dotPrinter(subsets);
+		*/
```

Comment:
Uncommented code should be removed. If you really need that code, add a debug switch (see [`PRINT_PEA_DOT`](https://github.com/ultimate-pa/ultimate/blob/c3cc0b0579c8badb9538468afd6bd266138627c8/trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/generator/RtInconcistencyConditionGenerator.java#L307)):

```suggestion
```